File: system/monolith/java/src/main/java/com/mycompany/myshop/monolith/core/dtos/BrowseCouponsResponse.java
package com.mycompany.myshop.monolith.core.dtos;

import lombok.Data;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.List;

@Data
public class BrowseCouponsResponse {
    private List<BrowseCouponsItemResponse> coupons;

    @Data
    public static class BrowseCouponsItemResponse {
        private String code;
        private BigDecimal discountRate;
        private Instant validFrom;
        private Instant validTo;
        private Integer usageLimit;
        private Integer usedCount;
    }
}

File: system/monolith/dotnet/Core/Dtos/BrowseCouponsResponse.cs
namespace MyCompany.MyShop.Monolith.Core.Dtos;

public class BrowseCouponsResponse
{
    public List<BrowseCouponsItemResponse> Coupons { get; set; } = new();
}

public class BrowseCouponsItemResponse
{
    public string Code { get; set; } = null!;
    public decimal DiscountRate { get; set; }
    public DateTime? ValidFrom { get; set; }
    public DateTime? ValidTo { get; set; }
    public int? UsageLimit { get; set; }
    public int UsedCount { get; set; }
}
